vector/src/main/java/fr/gouv/tchap/activity/TchapRoomCreationActivity.java

```diff
@@ -124,6 +127,19 @@ public void initUiAndData() {
         switchPublicPrivateRoom.setChecked(false);
         mRoomParams.visibility = RoomState.DIRECTORY_VISIBILITY_PRIVATE;
         mRoomParams.preset = CreateRoomParams.PRESET_PRIVATE_CHAT;
+
+        // Prepare disable federation label by adding the hs domain of the current user.
+        String userHSDomain = mSession.getMyUserId().substring(mSession.getMyUserId().indexOf(":") + 1);
```

Comment:
You should externalize this code (and the lines below), in `DinsicUtils` for instance, to be able to reuse it (like in line 234) and unit test it.